param/__init__.py

```diff
@@ -282,6 +288,27 @@ def _get_min_max_value(min, max, value=None, step=None):
     return min, max, value
 
 
+def _deserialize_from_path(ext_to_routine, path, type_name):
+    """Call deserialization routine with path according to extension"""
+    if not os.path.isfile(path):
+        raise FileNotFoundError("'{}' does not exist or is not a file".format(path))
+    root, ext = os.path.splitext(path)
+    if ext in {'.gz', '.bz2', '.xz', '.zip'}:
+        # A compressed type. We'll assume the routines can handle such extensions
+        # transparently
+        ext = os.path.splitext(root)[1]
+    # If the extension does match, this is likely the correct routine and we will error
+    # later on. Let's warn the user.
+    try:
+        if ext in ext_to_routine:
+            return ext_to_routine[ext](path)
+    except:
+        get_logger().warning(
+            "Could not parse file '{}' as {}".format(path, type_name), exc_info=True)
+    raise ValueError(
```

Comment:
Here might be a better point for a warning stating the possible extensions. At this point we've verified that the path is indeed a file on disk, but it wasn't listed in our extension list. On that note, we can use the keys of the dictionary for the list of supported extensions

``` python
get_logger().warning("Cannot parse file '{}' as {}: unknown extension '{}'. Supported extensions: {}".format(path, type_name, ext, ', '.join(sorted(ext_to_routine))
```

Keeping track of the supported extensions somewhere globally could also make sense. Where would we put it?

Reply:
> Keeping track of the supported extensions somewhere globally could also make sense. Where would we put it?

It could just be a global variable at the top level of the namespace. The tricky bit is naming it!

Reply:
My vote is that a global dictionary like that is too tricky due to imports, and would either involve hackery (eval, etc.) or would be out of sync with what the code actually does. But I do think we should list the valid extensions.

Reply:
Ok, I've updated the messages. They also error instead of warning as per our other discussions. Right now the list of supported extensions isn't in some global list - I'll leave you two to come to a consensus about that.

Reply:
The import issue does make it tricky so I guess leaving the listing implicit (i.e. as it is now) is fine.